### PR TITLE
[FIX] account[_peppol]: Delay link_documents after _call_web_service_…

### PR DESCRIPTION
**Pull request by smetl**
…after_invoice_pdf_render

If a web-service after the PDF generation failed but does a cr.commit(), the attachments should not be generated.

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by smetl** [comment]
@aliyatastemirova To check the impact of this PR on peppol

**Comment by aliyatastemirova** [comment]
@smetl I reverted the changes in the account_peppol module.
Your fix only concerns cases when the call to the web service fails and we know about it immediately. If the user tries to resend the invoice, we will have `invoice_data['ubl_cii_xml_attachment_values']` again.
However, if the call to the iap server succeeds, but there is an error when sending the message via peppol, we find out about it much later after the cron is run. Then, the peppol status changes to `error`. If the user wants to resend it, there will be no `invoice_data['ubl_cii_xml_attachment_values']` but we will have `invoice.ubl_cii_xml_id`, because the initial call was successful.

TL;DR: this change does not concern account_peppol

**Comment by jco-odoo** [comment]
@robodoo r+
